includes/Modules/Sign_In_With_Google/Tag_Guard.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Sign_In_With_Google\Tag_Guard
+ *
+ * @package   Google\Site_Kit\Modules\Sign_In_With_Google
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Sign_In_With_Google;
+
+use Google\Site_Kit\Core\Modules\Module_Settings;
+use Google\Site_Kit\Core\Modules\Tags\Module_Tag_Guard;
+
+/**
+ * Class for the Sign in with Google tag guard.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Tag_Guard extends Module_Tag_Guard {
+	/**
+	 * Determines whether the guarded tag can be activated or not.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool|WP_Error TRUE if guarded tag can be activated, otherwise FALSE or an error.
+	 */
+	public function can_activate() {
+		$settings = $this->settings->get();
+
+		// If there's no client ID available, don't render the button.
+		if ( ! $settings['clientID'] ) {
+			return false;
+		}
+
+		// If the site does not use https, don't render the button.
+		if ( substr( wp_login_url(), 0, 5 ) !== 'https' ) {
+			return false;
+		}
```

Comment:
This should be a separate guard. RRM has the same condition I believe, maybe there is already one that can be used here?

Also, I believe it is possible for https to be set indifferently on the login URL. This should probably be specific to the current request because this limitation is about the current page. We should also keep this check on the login URL specifically though because that is the endpoint we make the login request to, so both are important.

Reply:
That sounds like a good idea to me. Regarding the others, I don't think those are needed for SiwG. The verify guard suppresses the tag output when we are checking the page for "existing tags" which isn't something we do for SiwG currently. 

The environment type guard is something that is also more relevant for other tags (e.g. measurement or ads related) where you wouldn't want them to be active outside of production. I don't think that's the case for SiwG either – it either wouldn't work due to mismatching domains for the client ID, or it would, regardless of the env type.

Good questions though. Let's follow up to extract that separate guard class as you said in a new issue.